### PR TITLE
docs(react-native): add  code examples for common use cases

### PR DESCRIPTION
**Pull request by arnaudbzn**
When using React Query with React Native you often have to handle the following use cases:
- Online status management
- Refetch on App focus
- Refetch on Screen focus

The React Native page now includes ready to use examples for these common cases.

### CODE REVIEW

File: docs/src/pages/react-native.md [hunk 1]
```diff
@@ -3,8 +3,78 @@ id: react-native
...
+```ts
+import React from 'react'
+import NetInfo from '@react-native-community/netinfo'
+import { onlineManager } from 'react-query'
+import { Platform } from 'react-native'
+
+export function useOnlineManager() {
+  React.useEffect(() => {
+    if (Platform.OS !== 'web') {
+      return NetInfo.addEventListener(state => {
```
Comment (TkDodo) [inline docs/src/pages/react-native.md:26]: hmm, the [onlinemanager docs](https://react-query.tanstack.com/reference/onlineManager#onlinemanagerseteventlistener) have a slightly different example:

```
 import NetInfo from '@react-native-community/netinfo'
 import { onlineManager } from 'react-query'
 
 onlineManager.setEventListener(setOnline => {
   return NetInfo.addEventListener(state => {
     setOnline(state.isConnected)
   })
 })
```

File: docs/src/pages/react-native.md [hunk 2]
```diff
@@ -3,8 +3,78 @@ id: react-native
...
+In React Native you have to use React Query `focusManager` to refetch when the App is focused.
+You can use 'react-native-appstate-hook' to be notified when the App state has changed.
+
+```ts
+import { focusManager } from 'react-query'
+import useAppState from 'react-native-appstate-hook'
+
+function onAppStateChange(status: AppStateStatus) {
+  if (Platform.OS !== 'web') {
+    focusManager.setFocused(status === 'active')
```
Comment (TkDodo) [inline docs/src/pages/react-native.md:49]: same question, see [focusManager.setEventListener](https://react-query.tanstack.com/reference/focusManager#focusmanagerseteventlistener)

**Review by TkDodo — CHANGES_REQUESTED** [review]

File: docs/src/pages/react-native.md (hunk 1 above)
Comment (arnaudbzn) [reply docs/src/pages/react-native.md]: Agree, we can keep the same example.

File: docs/src/pages/react-native.md (hunk 2 above)
Comment (arnaudbzn) [reply docs/src/pages/react-native.md:39]: The example I provide is a React Native based solution using `react-native-appstate-hook` compared to the existing `setEventListener` documentation using `window`.

**Review by TkDodo — APPROVED** [review]

**Pull request merged**